experiments/ppo_gridnet.py

```diff
@@ -236,6 +237,66 @@ def get_value(self, x):
         return self.critic(self.encoder(x))
 
 
+def run_evaluation(model_path: str, output_path: str):
+    args = [
+        "python",
+        "league.py",
+        "--evals",
+        model_path,
+        "--update-db",
+        "false",
+        "--cuda",
+        "false",
+        "--output-path",
+        output_path
+    ]
+    fd = subprocess.Popen(args)
+    print(f"Evaluating {model_path}")
+    return_code = fd.wait()
+    assert return_code == 0
+    return (output_path, model_path)
+
+
+class TrueskillWriter:
+
+    def __init__(self, prod_mode, writer, league_path: str, league_step_path: str):
+        self._prod_mode = prod_mode
+        self._writer = writer
+        self._trueskill_df = pd.read_csv(league_path)
+        self._trueskill_step_df = pd.read_csv(league_step_path)
+        self._trueskill_step_df["type"] = self._trueskill_step_df["name"]
+        self._trueskill_step_df["step"] = 0
+        # xxx(okachaiev): not sure we need this copy
+        self._preset_trueskill_step_df = self._trueskill_step_df.copy()
+
+    def on_evaluation_done(self, future):
+        if not future.done(): return
```

Comment:
The future should be by definition done, right?

Reply:
It also can be canceled, or failed. I need to double check docs regarding my intuition but I believe that in case of cancelation, done should return false

Reply:
Oh I see; then this makes sense